diff -Nuar gcc-7.3.0/libsanitizer/asan/asan_rtl.cc gcc-7.3.0-patched/libsanitizer/asan/asan_rtl.cc
--- gcc-7.3.0/libsanitizer/asan/asan_rtl.cc	2018-06-11 00:32:39.616794037 +0300
+++ gcc-7.3.0-patched/libsanitizer/asan/asan_rtl.cc	2018-06-11 00:34:26.543540432 +0300
@@ -68,8 +68,8 @@
 }
 
 // -------------------------- Globals --------------------- {{{1
-int asan_inited;
-bool asan_init_is_running;
+int asan_inited = 0;
+bool asan_init_is_running = 0;
 
 #if !ASAN_FIXED_MAPPING
 uptr kHighMemEnd, kMidMemBeg, kMidMemEnd;
