diff -Nuar gcc-7.3.0/libgcc/unwind-dw2-fde.h gcc-7.3.0-patched/libgcc/unwind-dw2-fde.h
--- gcc-7.3.0/libgcc/unwind-dw2-fde.h	2017-01-01 15:07:43.905435000 +0300
+++ gcc-7.3.0-patched/libgcc/unwind-dw2-fde.h	2018-06-11 19:06:05.978617608 +0300
@@ -168,6 +168,9 @@
 static inline int
 last_fde (struct object *obj __attribute__ ((__unused__)), const fde *f)
 {
+  if (f->length >= 0xfffffff0) /* special DWARF extension */
+    return 1;
+
 #ifdef DWARF2_OBJECT_END_PTR_EXTENSION
   return f == (const fde *) obj->fde_end || f->length == 0;
 #else
