diff --git a/libgcc/Makefile.in b/libgcc/Makefile.in
index dd8cee9..f156398 100644
--- a/libgcc/Makefile.in
+++ b/libgcc/Makefile.in
@@ -41,7 +41,7 @@ long_double_type_size = @long_double_type_size@
 decimal_float = @decimal_float@
 enable_vtable_verify = @enable_vtable_verify@
 enable_decimal_float = @enable_decimal_float@
-fixed_point = @fixed_point@
+#fixed_point = @fixed_point@
 with_aix_soname = @with_aix_soname@
 enable_execute_stack = @enable_execute_stack@
 unwind_header = @unwind_header@
