diff -Nuar ncurses-6.0/Ada95/configure ncurses-6.0-patched/Ada95/configure
--- ncurses-6.0/Ada95/configure	2017-12-24 16:12:20.719375595 +0300
+++ ncurses-6.0-patched/Ada95/configure	2017-12-24 16:13:13.247195141 +0300
@@ -1601,13 +1601,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-Os"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-Os"
   else
     CFLAGS=
   fi
diff -Nuar ncurses-6.0/Ada95/package/debian/rules ncurses-6.0-patched/Ada95/package/debian/rules
--- ncurses-6.0/Ada95/package/debian/rules	2017-12-24 16:12:20.723375581 +0300
+++ ncurses-6.0-patched/Ada95/package/debian/rules	2017-12-24 16:13:13.259195102 +0300
@@ -19,7 +19,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
diff -Nuar ncurses-6.0/Ada95/src/library.gpr ncurses-6.0-patched/Ada95/src/library.gpr
--- ncurses-6.0/Ada95/src/library.gpr	2017-12-24 16:12:20.723375581 +0300
+++ ncurses-6.0-patched/Ada95/src/library.gpr	2017-12-24 16:13:13.263195088 +0300
@@ -47,7 +47,7 @@
   package Compiler is
      for Default_Switches ("Ada") use
        ("-g",
-        "-O2",
+        "-Os",
         "-gnatafno",
         "-gnatVa",   -- All validity checks
         "-gnatwa");  -- Activate all optional errors
diff -Nuar ncurses-6.0/configure ncurses-6.0-patched/configure
--- ncurses-6.0/configure	2017-12-24 16:12:20.707375638 +0300
+++ ncurses-6.0-patched/configure	2017-12-24 16:13:13.371194725 +0300
@@ -1952,13 +1952,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-Os"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-Os"
   else
     CFLAGS=
   fi
@@ -2827,13 +2827,13 @@
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-Os"
   else
     CXXFLAGS="-g"
   fi
 else
   if test "$GXX" = yes; then
-    CXXFLAGS="-O2"
+    CXXFLAGS="-Os"
   else
     CXXFLAGS=
   fi
diff -Nuar ncurses-6.0/package/debian/rules ncurses-6.0-patched/package/debian/rules
--- ncurses-6.0/package/debian/rules	2017-12-24 16:12:20.731375552 +0300
+++ ncurses-6.0-patched/package/debian/rules	2017-12-24 16:13:13.571194050 +0300
@@ -39,7 +39,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
diff -Nuar ncurses-6.0/package/debian-mingw/rules ncurses-6.0-patched/package/debian-mingw/rules
--- ncurses-6.0/package/debian-mingw/rules	2017-12-24 16:12:20.731375552 +0300
+++ ncurses-6.0-patched/package/debian-mingw/rules	2017-12-24 16:13:13.575194036 +0300
@@ -56,7 +56,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
diff -Nuar ncurses-6.0/package/debian-mingw64/rules ncurses-6.0-patched/package/debian-mingw64/rules
--- ncurses-6.0/package/debian-mingw64/rules	2017-12-24 16:12:20.731375552 +0300
+++ ncurses-6.0-patched/package/debian-mingw64/rules	2017-12-24 16:13:13.571194050 +0300
@@ -56,7 +56,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
diff -Nuar ncurses-6.0/test/configure ncurses-6.0-patched/test/configure
--- ncurses-6.0/test/configure	2017-12-24 16:12:20.719375595 +0300
+++ ncurses-6.0-patched/test/configure	2017-12-24 16:13:13.755193430 +0300
@@ -1706,13 +1706,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-Os"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-Os"
   else
     CFLAGS=
   fi
diff -Nuar ncurses-6.0/test/package/debian/rules ncurses-6.0-patched/test/package/debian/rules
--- ncurses-6.0/test/package/debian/rules	2017-12-24 16:12:20.719375595 +0300
+++ ncurses-6.0-patched/test/package/debian/rules	2017-12-24 16:13:13.795193295 +0300
@@ -23,7 +23,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
diff -Nuar ncurses-6.0/test/package/debian-mingw/rules ncurses-6.0-patched/test/package/debian-mingw/rules
--- ncurses-6.0/test/package/debian-mingw/rules	2017-12-24 16:12:20.719375595 +0300
+++ ncurses-6.0-patched/test/package/debian-mingw/rules	2017-12-24 16:13:13.799193281 +0300
@@ -24,7 +24,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
diff -Nuar ncurses-6.0/test/package/debian-mingw64/rules ncurses-6.0-patched/test/package/debian-mingw64/rules
--- ncurses-6.0/test/package/debian-mingw64/rules	2017-12-24 16:12:20.719375595 +0300
+++ ncurses-6.0-patched/test/package/debian-mingw64/rules	2017-12-24 16:13:13.795193295 +0300
@@ -24,7 +24,7 @@
 ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
         CFLAGS += -O0
 else
-        CFLAGS += -O2
+        CFLAGS += -Os
 endif
 ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
         INSTALL_PROGRAM += -s
