diff -Nuar gdb-8.1/gdb/dwarf2read.c gdb-8.1-patched/gdb/dwarf2read.c
--- gdb-8.1/gdb/dwarf2read.c	2018-01-31 05:58:50.000000000 +0300
+++ gdb-8.1-patched/gdb/dwarf2read.c	2018-06-03 19:08:34.191923501 +0300
@@ -26085,7 +26085,7 @@
     gdb_assert (m_abbrev_table.empty ());
     const size_t name_count = m_name_to_value_set.size ();
     m_bucket_table.resize
-      (std::pow (2, std::ceil (std::log2 (name_count * 4 / 3))));
+      (std::pow (2, std::ceil (log2 (name_count * 4 / 3))));
     m_hash_table.reserve (name_count);
     m_name_table_string_offs.reserve (name_count);
     m_name_table_entry_offs.reserve (name_count);
